### DISCUSSION TITLE
Need help: FormatException: Unexpected character (at character 1) <html>

### DISCUSSION THREAD

**Discussion by mwakaambrose** [pocketbase/pocketbase#1190] [Q&A] [answered]
Here is a simple file upload test that I am using to explore pocketbase

```dart
final body = <String, dynamic>{
        "remoteUserID": "test",
        "localTaskID": "1233"
      };

      final record = await pb.collection('tasks').create(
        body: body,
        files: [
          await http.MultipartFile.fromPath(
            "photo",
            "/data/user/0/com.example.kijani/cache/image_picker6439133946021991280.jpg",
          ),
        ],
      );
```

I get this error

```bash
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: ClientException: {url: http://kijani.theonehq.com/api/collections/tasks/records, isAbort: false, statusCode: 0, response: {}, originalError: FormatException: Unexpected character (at character 1)
E/flutter (11375): <html>
E/flutter (11375): ^
E/flutter (11375): }
E/flutter (11375): #0      PocketBase.send
package:pocketbase/src/client.dart:186
E/flutter (11375): <asynchronous suspension>
E/flutter (11375): #1      _TestScreenState.triggerIt
package:kijani/screens/test.dart:82
E/flutter (11375): <asynchronous suspension>
E/flutter (11375):
```

I don't see any logs on my pocketbase instance

Any pointer is much appreciated

**Comment by ganigeorgiev** [comment, marked as answer]
What versions of PocketBase and the Dart SDK you are using?

The `<html>` tag in the error indicates that a non-json response was returned for some reason. PocketBase always will return json (or no body at all for 204).

My guess is that you probably have a reverse proxy (eg. nginx) or something else that overwrite the default PocketBase response. Based on the url in the ClientException, it is also possible that this could be caused by http->https redirect.

**Reply by mwakaambrose** [reply]
Dart version 2.18.2
Flutter version 3.3.4

Yes: I do have an nginx reverse proxy, from guide [here](https://pocketbase.io/docs/going-to-production/)
No: I don't have https setup and all requests are being sent to http and I don't have https redirect setup

Also the server for pocketbase is up and I can access the admin area

**Reply by ganigeorgiev** [reply]
I'm not able to reproduce the issue locally (PocketBase v0.8 and v0.9 with PocketBase Dart SDK v0.5).

On what platform do you run the flutter app, web?

Could you also try making the request with just the map body without the files upload?

**Reply by mwakaambrose** [reply]
My app is on Android

After you hinted that it could be nginx related, I went to the server and tailed the nginx error logs to see if anything pops up and to my surprise I saw

```bash
2022/12/06 16:21:11 [error] 3340#3340: *922 client intended to send too large body: 2493492 bytes, client: 41.190.134.11, server: kijani.theonehq.com, request: "POST /api/collections/tasks/records HTTP/1.1", host: "kijani.theonehq.com"
```

I did a quick google search about what it meant and found out that I can increase the request size allowed. So I did that

```bash
client_max_body_size 10M;
``` 
in the nginx.conf

Restarted nginx and it work fine. 

Thank you for the swift help and response. Cheers!

Edit: Here is the guide I used [https://www.cyberciti.biz/faq/fix-client-intended-to-send-too-large-body-xyz-bytes-in-nginx/](https://www.cyberciti.biz/faq/fix-client-intended-to-send-too-large-body-xyz-bytes-in-nginx/)

**Reply by ganigeorgiev** [reply]
Good to know. I suspected that it could be `client_max_body_size` (btw. note that it exists in the example nginx template - https://pocketbase.io/docs/going-to-production/#using-reverse-proxy, but it is defined at the server level block).

**Reply by mwakaambrose** [reply]
Oh, I completely missed that. I only copied the location block for my setup. Great docs you got there 💪🏾 How do I mark this as answered?
